receiver/statsdreceiver/transport/udp_server.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net"
+	"strings"
+	"sync"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/protocol"
+)
+
+type udpServer struct {
+	wg         sync.WaitGroup
+	packetConn net.PacketConn
+	reporter   Reporter
+}
+
+var _ (Server) = (*udpServer)(nil)
+
+// NewUDPServer creates a transport.Server using UDP as its transport.
+func NewUDPServer(addr string) (Server, error) {
+	packetConn, err := net.ListenPacket("udp", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	u := udpServer{
+		packetConn: packetConn,
+	}
+	return &u, nil
+}
+
+func (u *udpServer) ListenAndServe(
+	parser protocol.Parser,
+	nextConsumer consumer.MetricsConsumerOld,
+	reporter Reporter,
+) error {
+	if parser == nil || nextConsumer == nil || reporter == nil {
+		return errNilListenAndServeParameters
+	}
+
+	u.reporter = reporter
+
+	buf := make([]byte, 65527) // max size for udp packet body (assuming ipv6)
+	for {
+		n, _, err := u.packetConn.ReadFrom(buf)
```

Comment:
Is there a difference vs just `Read` without ignoring the return value?

Reply:
I don't think `net.PacketConn` has a `Read` method, only `ReadFrom`.

Reply:
Thanks mixed it up with https://golang.org/pkg/net/#UDPConn.Read